src/component/directive.ts

```diff
@@ -271,13 +271,39 @@ QWeb.addDirective({
     let tattClass = node.getAttribute("t-att-class");
     let styleAttr = node.getAttribute("style");
     let tattStyle = node.getAttribute("t-att-style");
+
+    const dataAttrs = [...node.attributes].filter(a => /^(t-attf?-)?data-.+$/.test(a.name)).sort();
+    if (dataAttrs.length) {
+      const dashToCamel = (str: string) => {
+        return str.toLowerCase()
+          .replace(/(\-[a-z])/g, val => val.toUpperCase().replace('-', ''));
+      };
+      ctx.addLine("let dataAttrMap = {};")
+      let attrName = "", attrValue = "";
+      for (const dataAttr of dataAttrs) {
+        if (dataAttr.name.startsWith("t-att-")) {
+          attrName = dashToCamel(dataAttr.name.slice(11));
+          let v = ctx.getValue(dataAttr.value);
+          attrValue = typeof v === "string" ? ctx.formatExpression(v) : `scope.${v.id}`;
+        } else if (dataAttr.name.startsWith("t-attf-")) {
+          attrName = dashToCamel(dataAttr.name.slice(12));
+          let v = ctx.getValue(dataAttr.value);
+          attrValue = typeof v === "string" ? ctx.interpolate(v) : `scope.${v.id}`;
+        } else {
+          attrName = dashToCamel(dataAttr.name.slice(5));
+          attrValue = '"' + dataAttr.value + '"';
+        }
+        ctx.addLine(`dataAttrMap.${attrName} = dataAttrMap.${attrName} ? dataAttrMap.${attrName} + " " + ${attrValue} : ${attrValue};`)
```

Comment:
Not sure it makes sense to combine (by concatenation) data values. It works well for classnames, but here I guess I wouldn't do that.

Reply:
I agree. We should probably log warnings when template define more than once the same data-* attribute.